tests/Spec/Nri/Ui/Highlighter.elm

```diff
@@ -0,0 +1,315 @@
+                |> ensureFocusOn "Pothos"
+                |> done
+    , test "expands selection one element to the right on shift + right arrow and highlight selected elements" <|
+        \() ->
+            Highlightable.initFragments Nothing "Pothos indirect light"
+                |> program marker
+                |> shiftRight
+                |> releaseShiftRight
+                |> ensureMarked [ "Pothos", "", "indirect" ]
+                |> shiftRight
+                |> releaseShiftRight
+                |> ensureMarked [ "Pothos", "", "indirect", "", "light" ]
+                |> shiftRight
+                |> releaseShiftRight
+                |> ensureMarked [ "Pothos", "", "indirect", "", "light" ]
+                |> done
+    , test "expands selection one element to the left on shift + left arrow and highlight selected elements" <|
+        \() ->
+            Highlightable.initFragments Nothing "Pothos indirect light"
+                |> program marker
+                |> rightArrow
+                |> rightArrow
+                |> shiftLeft
+                |> releaseShiftLeft
+                |> ensureMarked [ "indirect", "", "light" ]
+                |> shiftLeft
+                |> releaseShiftLeft
+                |> ensureMarked [ "Pothos", "", "indirect", "", "light" ]
+                |> shiftLeft
+                |> releaseShiftLeft
+                |> ensureMarked [ "Pothos", "", "indirect", "", "light" ]
+                |> done
+    , test "merges highlights" <|
+        \() ->
+            Highlightable.initFragments Nothing "Pothos indirect light"
+                |> program marker
+                |> ensureFocusOn "Pothos"
+                |> shiftRight
+                |> releaseShiftRight
+                |> ensureMarked [ "Pothos", "", "indirect" ]
+                |> ensureFocusOn "indirect"
+                |> rightArrow
+                |> ensureFocusOn "light"
+                |> shiftLeft
+                |> releaseShiftLeft
+                |> ensureMarked [ "Pothos", "", "indirect", "", "light" ]
+                |> done
+    , test "selects element on MouseDown and highlights selected element on MouseUp" <|
+        \() ->
+            Highlightable.initFragments Nothing "Pothos indirect light"
+                |> program marker
+                |> ensureFocusOn "Pothos"
+                |> mouseDown "Pothos"
+                |> mouseUp "Pothos"
+                |> ensureMarked [ "Pothos" ]
+                |> done
+    , test "selects element on MouseDown, expands selection on MouseOver, and highlights selected elements on MouseUp" <|
+        \() ->
+            Highlightable.initFragments Nothing "Pothos indirect light"
+                |> program marker
+                |> ensureFocusOn "Pothos"
+                |> mouseDown "Pothos"
+                |> mouseOver "indirect"
+                |> mouseUp "Pothos"
+                |> ensureMarked [ "Pothos", "", "indirect" ]
```

Comment:
❓ Why is the ensureMarked argument `[ "Pothos", "", "indirect" ]`, rather than `[ "Pothos", " ", "indirect" ]`? I think we want to highlight the space, right?